File: DSA/Aarohi-Sharma/Day80_q2.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    void dfs(TreeNode* root, int isLeft, int len, int& mx){
        if(!root){
            return;
        }
        mx = max(mx, len);
        if(isLeft){    
            dfs(root -> right, 0, len + 1, mx); 
            dfs(root -> left, 1, 1, mx); 
        }
        else{ 
            dfs(root -> left, 1, len + 1, mx); 
            dfs(root -> right, 0, 1, mx); 
        }
    }
    int longestZigZag(TreeNode* root) {
        int mx = 0;
        dfs(root -> left, 1, 1, mx); 
        dfs(root -> right, 0, 1, mx);
        return mx;
    }
};

File: DSA/Ishita-Nigam/Day80/Day80_Q2.java
//Count Complete Tree Nodes
class Day80_Q2 {

    public static class TreeNode {
        int val;
        TreeNode left, right;

        TreeNode(int x) {
            val = x;
        }
    }

    public int countNodes(TreeNode root) {
        if (root == null) {
            return 0;
        }
        int nl = countNodes(root.left);
        int nr = countNodes(root.right);

        return 1 + nl + nr;
    }

    public static void main(String[] args) {
        Day80_Q2 solution = new Day80_Q2();

        TreeNode root = new TreeNode(1);
        root.left = new TreeNode(2);
        root.right = new TreeNode(3);
        root.left.left = new TreeNode(4);
        root.left.right = new TreeNode(5);
        root.right.left = new TreeNode(6);

        int result = solution.countNodes(root);
        System.out.println("Number of nodes: " + result);
    }
}
